airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/streams.py

```diff
@@ -249,13 +250,20 @@ def get_updated_state(self, current_stream_state: MutableMapping[str, Any], late
         """
         survey_id = latest_record.get("survey_id")
         latest_cursor_value = latest_record.get(self.cursor_field)
+
         current_stream_state = current_stream_state or {}
         current_state = current_stream_state.get(survey_id) if current_stream_state else None
         if current_state:
             current_state = current_state.get(self.cursor_field)
         current_state_value = current_state or latest_cursor_value
         max_value = max(current_state_value, latest_cursor_value)
-        new_value = {self.cursor_field: max_value}
+
+        # API pulls records ON and after this date, so add one second to prevent duplicate responses
+        updated_max_value = (datetime.strptime(max_value, "%Y-%m-%dT%H:%M:%S+00:00") + timedelta(seconds=1)).strftime(
+            "%Y-%m-%dT%H:%M:%S+00:00"
+        )
```

Comment:
@danieldiamond what is the chance that adding 1 second can also cause lost of data?

Reply:
well i've confirmed that `+00:00` is always zero for all responses in survey monkey.

however, in thinking about this solution. i'm now thinking it might increment each sync even if there is no record to add.
e.g. 
- first second: add 5 records and update cursor field to the last `updated_at` timestamp + 1 second
- second sync immediately after: add **zero** records but the cursor field gets updated to timestamp + 1 second + 1 second